crates/control/src/role_assignment.rs

```diff
@@ -412,12 +412,6 @@ fn process_role_state_machine(
         }
     }
 
-    if let Some(message) = incoming_message {
-        if message.player_number == player_number {
-            return (current_role, false, team_ball);
-        }
-    }
```

Comment:
What is the reason to remove that? Is this no longer necessary?

Reply:
ah, is this duplicated to the check in the message filter?

Reply:
yes